### PR TITLE
fix: refactor findingLastBucket into a func

### PR DESCRIPTION
**Pull request by Elbehery**
This PR refactors finding a sub bucket logic into a helper func.

related to https://github.com/etcd-io/bbolt/issues/602

cc @ahrtr

### CODE REVIEW

**Comment by ahrtr** [comment]
Let's talk about the details on next Monday's meeting.

File: cmd/bbolt/main.go [hunk 1]
```diff
@@ -939,15 +939,10 @@ func (cmd *keysCommand) Run(args ...string) error {
...
-		if lastbucket == nil {
-			return berrors.ErrBucketNotFound
-		}
-		for _, bucket := range buckets[1:] {
-			lastbucket = lastbucket.Bucket([]byte(bucket))
-			if lastbucket == nil {
-				return berrors.ErrBucketNotFound
-			}
+		var lastbucket *bolt.Bucket
+		lastbucket, err = findLastBucket(tx, buckets)
```
Comment (ahrtr) [inline cmd/bbolt/main.go:943]:
```suggestion
		lastbucket, err := findLastBucket(tx, buckets)
```

File: cmd/bbolt/main.go [hunk 2]
```diff
@@ -1030,15 +1025,10 @@ func (cmd *getCommand) Run(args ...string) error {
...
```
Comment (ahrtr) [inline cmd/bbolt/main.go:1029]:
```suggestion
		lastbucket, err := findLastBucket(tx, buckets)
```

**Comment by ahrtr** [comment]
Sorry for the late response. Please also rebase this PR.

**Comment by Elbehery** [comment]
sure, i just started to take my breath, still sick :'(

**Comment by Elbehery** [comment]
@ahrtr updated 👍🏽

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks @Elbehery

**Pull request merged**